content/en/docs/ambient/architecture/index.md

```diff
@@ -33,9 +84,8 @@ However, depending on the destination's capabilities, different behavior will oc
 If the destination is also captured, or otherwise has Istio proxy capabilities (such as a sidecar), the request will be upgraded to an encrypted {{< gloss "HBONE" >}}HBONE tunnel{{< /gloss >}}.
 If the destination has a waypoint proxy, in addition to being upgraded to HBONE, the request will instead be forwarded to that waypoint.
 
-Note that in the case of a request to a `Service`, a specific endpoint will be selected to determine if it has a waypoint.
-However, if it *has* a waypoint, the request will be sent with a target destination of the `Service`, not the selected endpoint.
-This allows the waypoint to apply service-oriented policies to the traffic.
+Note that in the case of a request to a `Service`, if the service *has* a waypoint and the waypoint can capture `service` or `all` traffic, the request will be sent to its waypoint to enforce service-oriented policies to the traffic.
+Similarly, in the case of a request to a `Pod` IP, if the pod *has* a waypoint and the waypoint can capture `workload` or `all` traffic, the request will be sent to its waypoint to enforce pod-oriented policies to the traffic.
 In the rare case that a `Service` has a mix of waypoint enabled and non-enabled endpoints, some requests would be sent to a waypoint while other requests to the same service would not.
```

Comment:
This is incorrect, a 'Service' either has a waypoint or it does not atomically

I think you may be conflating the definition of K8S Service with the common parlance for a bunch of related worklaods being a 'service'. This case can only happen for 'pods' and to-workload traffic but it's pretty unlikely. If we were going to say anything it would be...

"Since it is possible to vary the labels associated with pods in a Deployment it is technically possible for some pods to *use* a waypoint while others do not. Users are generally recommended to avoid this advanced use-case:"

Reply:
Yes I was thinking about the service is labelled use-waypoint and some of the workloads belong to the service are labelled as use a different waypoint or use 0 waypoint.  I think this matters in the old design but not in the current design where it relies on traffic to service or workload to decide if waypoint enabled.

accepted your suggestions!